src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,199 @@
+     * values as the option assignment. Valid options include any client constants, as described in the PHP manual:
+     *  - http://php.net/manual/en/memcached.constants.php
+     *
+     * @param string|null $dsn
+     * @param array       $opts
+     * @param string|null $persistentId
+     *
+     * @return MemcachedAdapter
+     */
+    public static function create($dsn = null, array $opts = array(), $persistentId = null)
+    {
+        $adapter = new static(new \Memcached($persistentId));
+        $adapter->setup($dsn ? array($dsn) : array(), $opts);
+
+        return $adapter;
+    }
+
+    /**
+     * @param string[] $dsns
+     * @param mixed[]  $opts
+     *
+     * @return bool
+     */
+    public function setup(array $dsns = array(), array $opts = array())
+    {
+        $isSuccess = true;
+        foreach ($opts as $opt => $val) {
+            $isSuccess = $this->setOption($opt, $val) && $isSuccess;
+        }
+        foreach ($dsns as $dsn) {
+            $isSuccess = $this->addServer($dsn) && $isSuccess;
+        }
+
+        return $isSuccess;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        return $this->client->setMulti($values, $lifetime)
+            && $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        return $this->client->getMulti($ids);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
+            || $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids)
+    {
+        $toDelete = count($ids);
+        foreach ((array) $this->client->deleteMulti($ids) as $result) {
+            if (true === $result || \Memcached::RES_NOTFOUND === $result) {
+                --$toDelete;
+            }
+        }
+
+        return 0 === $toDelete;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace)
+    {
+        if (!isset($namespace[0]) || false === $ids = $this->getIdsByPrefix($namespace)) {
+            return $this->client->flush();
+        }
+
+        $isCleared = true;
+        do {
+            $isCleared = $this->doDelete($ids) && $isCleared;
+        } while ($ids = $this->getIdsByPrefix($namespace));
+
+        return $isCleared;
+    }
+
+    private function getIdsByPrefix($namespace)
+    {
+        if (false === $ids = $this->client->getAllKeys()) {
+            return false;
+        }
+
+        return array_filter((array) $ids, function ($id) use ($namespace) {
+            return 0 === strpos($id, $namespace);
+        });
+    }
+
+    private function setOption($opt, $val)
+    {
+        $toRestore = error_reporting(~E_ALL);
```

Comment:
See https://github.com/symfony/symfony/pull/20863/files#diff-ed071d63655500d46a7c7866cdb74cddR193 for test checking a false return on error instead of an E_WARNING. Is this something you don't think we should do?